File: 1137.n-th-tribonacci-number.ts
// @leet start
function tribonacci(n: number): number {
  let arr = [0, 1, 1];
  if (n < 3) return arr[n];
  for (let i = 3; i <= n; i++) {
    arr = [arr[1], arr[2], arr[0] + arr[1] + arr[2]];
  }
  return arr[2];
};
// @leet end

File: 1137.n-th-tribonacci-number.go
// @leet start
func tribonacci(n int) int {
  res := []int{0,1,1}
  if n < 3 { return res[n] }
  for i := 3; i <= n; i++ {
    res = []int{res[1], res[2], res[0] + res[1] + res[2]}
  }
  return res[2]
}
// @leet end
